snyk-container-scan-policy-ts/README.md

```diff
@@ -0,0 +1,71 @@
+# snyn-container-scan-policy
+
+Scan Pulumi-managed Docker containers with Snyk and Pulumi Policy as Code:
+
+- The code in the `infra` directory creates two `docker.Image` resources:
+    1. An image sourced from the `alpine` image, which does not have critical vulnerabilities.
+    1. An image sourced from the `debian` image, which has critical vulnerabilities.
+- The code in the `policy` directory contains a Pulumi Policy Pack which calls the Snyk CLI. If the Snyk CLI fails (e.g. because it detects vulnerabilities), the resource will be considered in violation and the `pulumi preview` (or `pulumi up`) operation will fail.
+
+To run the demo:
+
+```bash
+cd infra
+pulumi preview --policy-pack ../policy
+```
+
+## Configuration Options
+
+The `snyk-container-scan` policy has the following configurable options. These can be set by altering values in `policy-config.json` and calling Pulumi CLI with the `--policy-config` option, e.g.:
+
+```bash
+cd infra
+pulumi preview --policy-pack ../policy --policy-pack-config policy-config.json
```

Comment:
When I run this, I get the following error. Expected?

```
Previewing update (dev)

View in Browser (Ctrl+O): https://app.pulumi.com/christian-pulumi-corp/snyk-container-scan-policy-ts/dev/previews/45a17a21-3daf-45f6-abd9-38f60d8de683

Loading policy packs...

     Type                   Name                               Plan       Info
 +   pulumi:pulumi:Stack    snyk-container-scan-policy-ts-dev  create     1 error; 2 messages
 +   ├─ docker:index:Image  alpine                             create     
 +   └─ docker:index:Image  debian                             create     1 error

Policies:
    ✅ snyk-container-scanning@v0.0.1 (local: ../policy)

Diagnostics:
  pulumi:pulumi:Stack (snyk-container-scan-policy-ts-dev):
    (node:11647) DeprecationWarning: Calling start() is no longer necessary. It can be safely omitted.
    (Use `node --trace-deprecation ...` to show where the warning was created)

    error: preview failed

  docker:index:Image (debian):
    error: Preview failed: error reading build output: failed to register layer: write /var/cache/debconf/templates.dat: no space left on device
```